### DISCUSSION TITLE
Websocket Timeout .

### DISCUSSION THREAD

**Discussion by adityakrk** [quarkusio/quarkus#41575] [Q&A]
Hi ,
I am implemented websocket api to establish the connection with clients . Basically i want websocket to be alive until clients force to close the connection so i have implemented ping-pong functionality to keep websocket alive. But websocket got disconnects after 15-20 min even after having ping-pong functionality . Here is the dependency and version which i have used .
```
`implementation 'io.quarkus:quarkus-websockets'`
version : 3.4.1
```
can you tell me is this the excepted behavior of websocket ? if not can you suggest me how to handle this kind of scenario .

**Comment by cescoffier** [comment]
Did you try with `websocket-next`? https://quarkus.io/guides/websockets-next-tutorial

**Reply by adityakrk** [reply]
Actually we won't be using websocket-next as per the requirement . Is there any way to make it work using this dependency implementation 'io.quarkus:quarkus-websockets.

**Reply by adityakrk** [reply]
Is this the excepted behavior of websocket?

**Reply by cescoffier** [reply]
It depends on the infrastructure. The infra may recycle the connection. Also, you may need to check the TCP ideal timeout and ensure the ping/pong frames are sent. 

In general, when the infrastructure is a bit overzealous, it's better to send "application frames" (like an empty JSON object) periodically. Yes, it requires a bit of code on both sides, but it's by far the most reliable.

In WebSocket Next, we implemented the ping/pong in a way that works with this overzealous infra. I don't know about quarkus websocket.

**Reply by adityakrk** [reply]
Actually i am using nginx ingress to connect with websocket endpoint . I saw there is a keepalive_timeout = 10min for ingress.  May be i feel like this is the reason our websocket will disconnect after some minutes (approx 20min) and it is giving me 
`1006 Abnormal Closure:No close frame was received. `
